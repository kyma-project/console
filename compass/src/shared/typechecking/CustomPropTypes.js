//TODO: move this component to a shared "place"
class CustomPropTypes {
  static elementRef = (props, propName, componentName) =>
    props[propName] && props[propName].current !== undefined
      ? null
      : new Error(
          'Invalid prop `' +
            propName +
            '` supplied to' +
            ' `' +
            componentName +
            '`. Validation failed.',
        );

  static oneOfProps = (props, componentName, propsToLookFor = []) => {
    if (!propsToLookFor.some(propName => props[propName])) {
      return new Error(
        `One of props [${propsToLookFor.join(
          ', ',
        )}] was not specified in '${componentName}'.`,
      );
    }
  };
}

export default CustomPropTypes;
