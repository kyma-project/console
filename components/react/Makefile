ci-pr: resolve build validate test
ci-master: resolve build validate test
ci-release: resolve build validate test

resolve:
	npm install

build:
	npm run build

validate:
	npm run lint-check
	eslint -c .eslinterrc ./src

test:
	CI=true npm run test
	