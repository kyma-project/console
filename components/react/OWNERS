reviewers:
    - default-reviewers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-reviewers