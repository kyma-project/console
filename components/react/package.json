{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "@babel/runtime-corejs2": "^7.12.5",
    "lodash": "^4.17.21",
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "exec-sh": "^0.3.3",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^24.0.0",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
