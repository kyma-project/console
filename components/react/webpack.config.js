const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');

require('babel-polyfill');

let libraryName = pkg.name;

module.exports = {
  entry: ['./src/index'],
  module: {
    loaders: [
      { test: /\.js?$/, loader: 'babel-loader', exclude: /node_modules/ },
      {
        test: /\.css$/,
        use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2|otf)$/,
        loader: 'file-loader?name=fonts/[name].[ext]',
      },
    ],
  },
  resolve: {
    modules: [path.resolve('./node_modules'), 'node_modules'],
    extensions: ['.js'],
  },
  output: {
    path: path.join(__dirname, '/lib'),
    publicPath: '/',
    filename: 'index.js',
    library: libraryName,
    libraryTarget: 'umd',
    umdNamedDefine: true,
  },
  devtool: 'source-map',
  devServer: {
    contentBase: './src',
    hot: true,
  },
  plugins: [
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
  ],
  externals: {
    'styled-components': {
      commonjs: 'styled-components',
      commonjs2: 'styled-components',
      amd: 'styled-components',
    },
  },
};
