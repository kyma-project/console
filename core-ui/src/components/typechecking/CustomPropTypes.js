import PropTypes from 'prop-types';

//TODO: move this component to a shared "place"
export const elementRef = PropTypes.shape({ current: PropTypes.element });

export default {
  elementRef,
};
