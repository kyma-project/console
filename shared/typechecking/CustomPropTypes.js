import PropTypes from 'prop-types';

export const ref = PropTypes.shape({ current: PropTypes.any });

export const button = function(props, propName, componentName) {
  if (!props[propName]) return false;

  const buttonProps = new Map([
    ['compact', 'boolean'],
    ['disabled', 'boolean'],
    ['glyph', 'string'],
    ['label', 'string'],
    ['text', 'string'],
  ]);
  function hasTextOrLabel() {
    return (
      !props[propName].hasOwnProperty('text') &&
      !props[propName].hasOwnProperty('label') &&
      new Error(`Either "text" or "label" is required`)
    );
  }

  function checkTypes() {
    for (const [key, type] of buttonProps.entries()) {
      if (
        props[propName].hasOwnProperty(key) &&
        typeof props[propName][key] !== type
      ) {
        return new Error(
          `Invalid prop ${propName} supplied to ${componentName}. Expected ${key} to be ${type}.`,
        );
      }
    }
  }

  const hasRequiredProps = hasTextOrLabel();
  if (hasRequiredProps) return hasRequiredProps || null;

  const hasPropsOfValidTypes = checkTypes();
  if (hasPropsOfValidTypes) return hasPropsOfValidTypes || null;

  return null;
};

export default {
  button,
  ref,
};
