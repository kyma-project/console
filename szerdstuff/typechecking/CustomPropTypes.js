import PropTypes from 'prop-types';

const CustomPropTypes = PropTypes.shape({ current: PropTypes.any });

export { CustomPropTypes };
