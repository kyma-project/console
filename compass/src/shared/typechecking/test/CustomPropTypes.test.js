import CustomPropTypes from '../CustomPropTypes';
import { createRef } from 'react';

describe('CustomPropTypes', () => {
  describe('elementRef', () => {
    it('Returns null for a valid ref', () => {
      const props = {
        testprop: createRef(null),
      };

      expect(
        CustomPropTypes.elementRef(props, 'testprop', 'testcomponent'),
      ).toBe(null);
    });

    it('Returns an error for string', () => {
      const props = {
        testprop: 'somestring',
      };

      expect(
        CustomPropTypes.elementRef(props, 'testprop', 'testcomponent'),
      ).toMatchSnapshot();
    });

    it('Returns an error for null', () => {
      const props = {
        testprop: 'somestring',
      };

      expect(
        CustomPropTypes.elementRef(props, 'testprop', 'testcomponent'),
      ).toMatchSnapshot();
    });
  });

  describe('oneOfProps', () => {
    const allProps = ['a', 'b', 'c'];

    it('Returns no error if one of properties is provided', () => {
      expect(
        CustomPropTypes.oneOfProps({ a: 123 }, 'componentA', allProps),
      ).toBe(undefined);
    });

    it('Returns no error all props are provided', () => {
      expect(
        CustomPropTypes.oneOfProps(
          { a: 123, b: 55, c: 6436 },
          'componentA',
          allProps,
        ),
      ).toBe(undefined);
    });

    it('Returns error no prop is provided', () => {
      expect(
        CustomPropTypes.oneOfProps({}, 'componentA', allProps),
      ).toMatchSnapshot();
    });
  });
});
