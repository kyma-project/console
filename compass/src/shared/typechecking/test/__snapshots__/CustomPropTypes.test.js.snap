// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPropTypes elementRef Returns an error for null 1`] = `[Error: Invalid prop \`testprop\` supplied to \`testcomponent\`. Validation failed.]`;

exports[`CustomPropTypes elementRef Returns an error for string 1`] = `[Error: Invalid prop \`testprop\` supplied to \`testcomponent\`. Validation failed.]`;

exports[`CustomPropTypes oneOfProps Returns error no prop is provided 1`] = `[Error: One of props [a, b, c] was not specified in 'componentA'.]`;
